.github/workflows/install-with-rancher.yaml

```diff
@@ -0,0 +1,81 @@
+name: Install ROS on a server through Rancher
+
+on:
+  workflow_dispatch:
+    inputs:
+      registry:
+        required: false
+        default: quay.io
+        type: string
+        description: Registry to use
+      org:
+        required: false
+        default: costoolkit
+        type: string
+        description: Organization to use
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_USERNAME: ${{ secrets.QUAY_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.QUAY_PASSWORD }}
+      DOCKER_REGISTRY: ${{ github.event.inputs.registry }}
+      REPO: ${{ github.event.inputs.registry }}/${{ github.event.inputs.org }}/os2-ci
+      REPO_ROS_OPERATOR: ${{ github.event.inputs.registry }}/${{ github.event.inputs.org }}/ros-operator-ci
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Make images (container/iso/qcow2/etc.)
+        run: make ci
+      - name: Upload results
+        uses: actions/upload-artifact@v3
+        with:
+          name: artifacts
+          path: dist
+          if-no-files-found: error
+      - name: Release space from worker ♻
+        if: always()
+        run: |
+          sudo rm -rf build bin dist
+          docker system prune -f -a --volumes
+  install-with-rancher:
+    runs-on: self-hosted
+    needs: build
+    container:
+      image: opensuse/leap:latest
+      env:
+        CLUSTER_NAME: cluster-k3s
+        CLUSTER_NS: fleet-default
+        K8S_VERSION: v1.21.9+k3s1
+        KUBECONFIG: /etc/rancher/k3s/k3s.yaml
+      options: --privileged
+    steps:
+      - name: Install dependencies
+        run: zypper -n in -l qemu-kvm libvirt virt-install sudo curl helm git-core tar
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: artifacts
+      - name: Install Rancher
+        run: scripts/install-rancher
```

Comment:
maybe we can improve that part and run rancher on either `k3d` or `kind`. In that way we can maybe also run the e2e tests directly in github action runners as well?

Reply:
AFAIK Rancher on K3d is not supported. And moreover if K3d/Rancher is run on a GHA runner I'm afraid that it would not be able to contact a self-hosted/cirrus-ci runner. Rancher can be directly run as a container, I first thinking to use this, but for the reason explained before I don't think that it could run.

Reply:
I was thinking that if we get ahead with k3d or kind (seems kind works on macOS too) and the vm directly on the host, at that point a self-hosted is not even needed because we could run it on vbox in macOS runner. but I'm not sure if kind/rancher works there

Reply:
The main issue for me to use macOS is that for now there is x86 macOS runner on GH, but for how long? As they are aarch64 based now ;-)